src/util/containers/SingleRootContainerManager.ts

```diff
@@ -0,0 +1,35 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { InternalServerError } from '../errors/InternalServerError';
+import { ensureTrailingSlash } from '../PathUtil';
+import type { ContainerManager } from './ContainerManager';
+
+/**
+ * A ContainerManager that assumes there is only 1 root and all other identifiers are made by appending to that root.
+ */
+export class SingleRootContainerManager implements ContainerManager {
```

Comment:
Alternatively:

```suggestion
export class SingleHostContainerManager implements ContainerManager {
```

to complement the future `MultiHost…` setup that will be the other implementation?

But I think `SingleRoot…` actually makes sense, because there's only one way of doing a single root. Whereas with multiple roots, there's more options.

So yeah, please leave as is for now, just wanted to mention this thought. Also might become `SingleRootIdentifierStrategy` as per my other comments.